### PR TITLE
Development

### PR DESCRIPTION
**Pull request by DerCalli**
Description

refactor as discussed

- Using single plugin for all session attributes
- Rename model to expander
- add unit-tests

### CODE REVIEW

File: src/FondOfSpryker/Shared/GoogleTagManagerSessionConnector/Transfer/google_tag_manager_session_connector.transfer.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
```
Comment (daniel-rose) [inline src/FondOfSpryker/Shared/GoogleTagManagerSessionConnector/Transfer/google_tag_manager_session_connector.transfer.xml:1]: Unused stuff in this package...

File: src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Plugin/DataLayer/SessionDataLayerExpanderPluginExpanderPlugin.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Plugin\DataLayer;
+
+use FondOfSpryker\Yves\GoogleTagManagerExtension\Dependency\GoogleTagManagerDataLayerExpanderPluginInterface;
+use Spryker\Yves\Kernel\AbstractPlugin;
+
+/**
+ * @method \FondOfSpryker\Yves\GoogleTagManagerSessionConnector\GoogleTagManagerSessionConnectorFactory getFactory()
+ */
+class SessionDataLayerExpanderPluginExpanderPlugin extends AbstractPlugin implements GoogleTagManagerDataLayerExpanderPluginInterface
```
Comment (julianhyatt) [inline src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Plugin/DataLayer/SessionDataLayerExpanderPluginExpanderPlugin.php]: SessionDataLayerExpanderPluginExpanderPlugin => SessionDataLayerExpanderPlugin

File: src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Expander;
+
+use FondOfSpryker\Shared\GoogleTagManagerSessionConnector\GoogleTagManagerSessionConnectorConstants;
+use FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Dependency\GoogleTagManagerSessionConnectorToSessionClientInterface;
+
+class SessionDataLayerExpander implements SessionDataLayerExpanderInterface
```
Comment (daniel-rose) [inline src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php]: SessionDataLayerExpander -> DataLayerExpander
SessionDataLayerExpanderInterface -> DataLayerExpanderInterface

File: src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        return $dataLayer;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getSessionId(): string
+    {
+        return $this->sessionClient->getId();
+    }
```
Comment (daniel-rose) [inline src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php]: remove this function

File: src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * @param string $page
+     * @param array $twigVariableBag
+     * @param array $dataLayer
+     *
+     * @return array
+     */
+    public function expand(string $page, array $twigVariableBag, array $dataLayer): array
+    {
+        $dataLayer[GoogleTagManagerSessionConnectorConstants::FIELD_TRANSACTION_ID] = $this->getSessionId();
```
Comment (daniel-rose) [inline src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php]: `$dataLayer[GoogleTagManagerSessionConnectorConstants::FIELD_TRANSACTION_ID] = $this->sessionClient->getId();`

File: src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/GoogleTagManagerSessionConnectorFactory.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+use FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Expander\SessionDataLayerExpander;
+use FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Expander\SessionDataLayerExpanderInterface;
+use Spryker\Yves\Kernel\AbstractFactory;
+
+class GoogleTagManagerSessionConnectorFactory extends AbstractFactory
+{
+    /**
+     * @return \FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Expander\SessionDataLayerExpanderInterface
+     */
+    public function createSessionDataLayerExpander(): SessionDataLayerExpanderInterface
```
Comment (daniel-rose) [inline src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/GoogleTagManagerSessionConnectorFactory.php]: createDataLayerExpander()

**Review by daniel-rose — CHANGES_REQUESTED** [review]

File: src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php (hunk 3 above)
Comment (DerCalli) [reply src/FondOfSpryker/Yves/GoogleTagManagerSessionConnector/Expander/SessionDataLayerExpander.php]: Maybe better move the interface to fond-of-spryker/google-tag-manager-extension?

File: src/FondOfSpryker/Shared/GoogleTagManagerSessionConnector/Transfer/google_tag_manager_session_connector.transfer.xml (hunk 1 above)
Comment (DerCalli) [reply src/FondOfSpryker/Shared/GoogleTagManagerSessionConnector/Transfer/google_tag_manager_session_connector.transfer.xml:1]: yesterday it was okay :D :D :D

**Review by daniel-rose — APPROVED** [review]

**Pull request merged**